File: backend/app/schemas/audit.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class AuditLogRead(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID | None = None
    user_email: str
    action: str
    bucket: str | None = None
    object_key: str | None = None
    details: dict | None = None
    ip_address: str | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class AuditPage(BaseModel):
    items: list[AuditLogRead]
    total: int
    page: int
    page_size: int

File: frontend/src/api/audit.ts
import apiClient from './client'

export interface AuditEntry {
  id: string
  user_id: string | null
  user_email: string
  action: string
  bucket: string | null
  object_key: string | null
  details: Record<string, any> | null
  ip_address: string | null
  created_at: string
}

export interface AuditPage {
  items: AuditEntry[]
  total: number
  page: number
  page_size: number
}

export interface AuditFilters {
  action?: string
  bucket?: string
  date_from?: string
  date_to?: string
  page?: number
  page_size?: number
}

export const auditApi = {
  list: (params: AuditFilters = {}) =>
    apiClient.get<AuditPage>('/audit', { params }),
}
